ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -199,9 +205,11 @@ private static void validateSchemaInference(
     if (!formatSupportsSchemaInference(formatInfo)) {
       final String msg = (isKey ? CommonCreateConfigs.KEY_FORMAT_PROPERTY
           : CommonCreateConfigs.VALUE_FORMAT_PROPERTY) + " should support schema inference "
-              + "when " + schemaIdName + " is provided!";
+          + "when " + schemaIdName + " is provided!";
```

Comment:
Can we add the format into this error message so the error message is more helpful to users? I think this is especially important for the `C*AS` case where we might be inheriting the format from the source. The user might not realize what format is being used otherwise.

Also (a nit), can we replace the exclamation mark with a period? Very few ksqlDB error messages use exclamation marks so this one reads a little strangely IMO.